src/odemis/driver/pigcs.py

```diff
@@ -1873,21 +1884,23 @@ def __init__(self, busacc, address=None, axes=None, auto_suspend=10):
             # To be taken when reading position or affecting encoder reading
             self._pos_lock[a] = threading.RLock()
 
-            # TODO: allow to force servo_suspend off on E861 with encoder off support
             # We have two modes for auto_suspend:
-            #  * Servo and encoder off: used when encoders can be turned off.
+            #  * full: Servo and encoder off.
             #    That typically happens with the C-867. It allows to reduce heat
             #    due to encoder using infra-red light (and ensures the motor
-            #    doesn't move.
-            #  * PID values set to 0,0,0: used when encoders cannot be turned off.
+            #    doesn't move).
+            #  * read: PID values set to 0,0,0.
             #    That typically happens with the E-861. It avoids going through
             #    the piezo "relax" procedure that takes time (4 x slew rate) and
-            #    causes up to 100 nm move.
-            if 0x56 in self._avail_params:  # Parameter to control encoder power
+            #    causes up to 100 nm move. It also allows to monitor the position.
+            if suspend_mode == "full":
                 self._servo_suspend = True
                 if self._auto_suspend:
                     logging.info("Will turn off servo when axis not in use")
-            else:
+                    if 0x56 not in self._avail_params:  # Parameter to control encoder power
+                        logging.warning("Controller %d cannot turn off sensor so suspend_mode full is not useful",
+                                        address)
+            elif suspend_mode == "read":
```

Comment:
Is it useful to have an else statement in case there is e.g. a typo in the yaml, so neither `read` nor `full` is True?

Reply:
That's a good catch. Indeed, that's the whole point of using "elif". I forgot to write the "else" block. Just added it now.